code/workspaces/auth-service/src/controllers/authorisation.js

```diff
@@ -43,6 +43,18 @@ function getPermissionsForUser(request, response) {
     });
 }
 
+async function getRolesForOtherUser(request, response) {
```

Comment:
It would be nice if this was named getRolesForUser and worked for both cases of working for the active user and also for others explictly, but can see why you've done it like this.

Reply:
I think it makes sense to validate the userId, as is done in routes.js:
`  app.get('/roles/:userId', dtMW, getUserValidator, permissionChecker(SYSTEM_INSTANCE_ADMIN), auth.getRolesForOtherUser);
`
This way if you mis-spell or omit userId, you know straight away, rather than it defaulting to the active user.